src/containers/Explorer/index.js

```diff
@@ -1,15 +1,40 @@
 import React from 'react';
 import FilterNav from './FilterNav';
 import ExplorerBody from './ExplorerBody';
+import { withRouter } from 'react-router-dom';
+import { compose } from 'redux';
+import { connect } from 'react-redux';
+import { actions } from 'public-modules/Bounties';
+import { toggleFromParam } from 'utils/locationHelpers';
 import styles from './Explorer.module.scss';
 
-const Explorer = props => {
-  return (
-    <div className={`${styles.explorerContainer}`}>
-      <FilterNav />
-      <ExplorerBody />
-    </div>
-  );
-};
+class Explorer extends React.Component {
+  constructor(props) {
+    super(props);
+    const { resetState, history, location, load, toggleStageFilter } = props;
+    resetState();
```

Comment:
It is now important that you call resetState within constructors on the pages that manage the loading logic. This makes sure EVERY time you are routed to the page, the data is fresh and clean. We should begin using this pattern everywhere.

Also - you'll see there is no need to manage batching here since we do not need to batch if the loaded flag has not been set to true yet :).